### PR TITLE
Update remacs build files

### PR DESCRIPTION
**Pull request by jman-schief**
This PR has some preliminary work on `remacs` for this repository. Things I've changed while exploring  issue #230 :
- removed deprecated pyml syntax
- updated rust nightly compiler
- updated README

I am unsure where to go next from here. Unclear points:

- The Dockerfiles seems to just build an image with the Rust compiler. I could not understand from the attached README how to compile remacs. I've edited the README with my best-effort guesswork on how to do that, but I probably need some input here :-)

 - The only script I've found to build remacs (in `rust_src/admin/build-and-run-remacs.sh`) fails the build, but is this script still in use?

<details><summary>Compile error</summary>

   Compiling swc_ecma_transforms_base v0.12.6
   Compiling swc_ecma_codegen v0.52.4
   Compiling swc_ecma_transforms_optimization v0.15.6
   Compiling swc_ecma_transforms_proposal v0.13.4
   Compiling swc_ecma_transforms_typescript v0.14.4
   Compiling swc_ecma_transforms_react v0.14.4
   Compiling deno_runtime v0.13.0 (https://github.com/DavidDeSimone/deno?branch=emacs-ng#64d58c91)
   Compiling swc_ecma_transforms v0.45.3
   Compiling swc_ecmascript v0.31.4
   Compiling dprint-swc-ecma-ast-view v0.15.0
   Compiling dprint-swc-ecma-ast-view v0.16.0
   Compiling deno_doc v0.3.0
   Compiling deno_lint v0.4.0
   Compiling dprint-plugin-typescript v0.44.2
   Compiling deno v1.9.2 (https://github.com/DavidDeSimone/deno?branch=emacs-ng#64d58c91)
    Finished release [optimized] target(s) in 9m 13s
gcc -o temacs.tmp \
  -Demacs  -I. -I. -I../lib -I../lib   -isystem /usr/include/gtk-3.0 -isystem /usr/include/pango-1.0 -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/harfbuzz -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/libmount -isystem /usr/include/blkid -isystem /usr/include/fribidi -isystem /usr/include/libxml2 -isystem /usr/include/cairo -isystem /usr/include/pixman-1 -isystem /usr/include/gdk-pixbuf-2.0 -isystem /usr/include/gio-unix-2.0 -isystem /usr/include/cloudproviders -isystem /usr/include/atk-1.0 -isystem /usr/include/at-spi2-atk/2.0 -isystem /usr/include/dbus-1.0 -isystem /usr/lib64/dbus-1.0/include -isystem /usr/include/at-spi-2.0 -pthread      -isystem /usr/include/libpng16  -isystem /usr/include/libxml2  -isystem /usr/include/dbus-1.0 -isystem /usr/lib64/dbus-1.0/include        -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/libmount -isystem /usr/include/blkid -pthread  -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -pthread  -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/harfbuzz -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -pthread  -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/harfbuzz -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -pthread -isystem /usr/include/libxml2  -isystem /usr/include/harfbuzz -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -pthread -isystem /usr/include/freetype2 -isystem /usr/include/libpng16    -MMD -MF deps/.d -MP   -isystem /usr/include/p11-kit-1   -isystem /usr/include/cairo -isystem /usr/include/glib-2.0 -isystem /usr/lib64/glib-2.0/include -isystem /usr/include/sysprof-4 -isystem /usr/include/pixman-1 -isystem /usr/include/freetype2 -isystem /usr/include/libpng16 -isystem /usr/include/harfbuzz -isystem /usr/include/libxml2 -pthread   -pthread -DEMACS_EXTERN_INLINE -fno-common -Wall -Warith-conversion -Wdate-time -Wdisabled-optimization -Wdouble-promotion -Wduplicated-cond -Wextra -Wformat-signedness -Winit-self -Winvalid-pch -Wlogical-op -Wmissing-declarations -Wmissing-include-dirs -Wmissing-prototypes -Wnested-externs -Wnull-dereference -Wold-style-definition -Wopenmp-simd -Wpacked -Wpointer-arith -Wstrict-prototypes -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wsuggest-final-methods -Wsuggest-final-types -Wtrampolines -Wuninitialized -Wunknown-pragmas -Wunused-macros -Wvariadic-macros -Wvector-operation-performance -Wwrite-strings -Warray-bounds=2 -Wattribute-alias=2 -Wformat=2 -Wformat-truncation=2 -Wimplicit-fallthrough=5 -Wshift-overflow=2 -Wvla-larger-than=4031 -Wredundant-decls -Wno-missing-field-initializers -Wno-override-init -Wno-sign-compare -Wno-type-limits -Wno-unused-parameter -Wno-format-nonliteral -O0 -g3   -L.././rust_src/target/release \
    dispnew.o frame.o scroll.o xdisp.o menu.o xmenu.o window.o charset.o coding.o category.o ccl.o character.o chartab.o bidi.o cm.o term.o terminal.o xfaces.o xterm.o xfns.o xselect.o xrdb.o xsmfns.o xsettings.o gtkutil.o emacsgtkfixed.o dbusbind.o emacs.o keyboard.o macros.o keymap.o sysdep.o bignum.o buffer.o filelock.o insdel.o marker.o minibuf.o fileio.o dired.o cmds.o casetab.o casefiddle.o indent.o search.o regex-emacs.o undo.o alloc.o pdumper.o data.o doc.o editfns.o callint.o eval.o floatfns.o fns.o font.o print.o lread.o emacs-module.o syntax.o  bytecode.o comp.o dynlib.o process.o gnutls.o callproc.o region-cache.o sound.o timefns.o atimer.o doprnt.o intervals.o textprop.o composite.o xml.o lcms.o inotify.o  profiler.o decompress.o thread.o systhread.o      xfont.o ftfont.o ftcrfont.o hbfont.o  fontset.o fringe.o image.o xgselect.o   -lemacsng terminfo.o lastfile.o      ../lib/libgnu.a       -ltiff -ljpeg -lpng16 -lz  -lgif -lXpm -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -lharfbuzz -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0  -lSM -lICE -lX11 -lX11-xcb -lxcb  -lXrender        -lrt -ldbus-1   -lXrandr  -lXinerama  -lXfixes  -lXext -lxml2    -lcairo  -ltinfo  -lgio-2.0 -lgobject-2.0 -lglib-2.0  -lgobject-2.0 -lglib-2.0  -lselinux -lfreetype  -lfontconfig -lfreetype  -lharfbuzz    -lgnutls  -lpthread -lanl   -lm -lz -ldl   -lgmp  -lX11-xcb -lxcb -ldl -lm -lrt -lstdc++
../lib-src/make-fingerprint temacs.tmp
mv temacs.tmp temacs
/usr/bin/mkdir -p ../etc
make -C ../lisp update-subdirs
make[2]: Entering directory '/root/emacs/lisp'
make[2]: Leaving directory '/root/emacs/lisp'
cp -f temacs bootstrap-emacs
rm -f bootstrap-emacs.pdmp
./temacs --batch  -l loadup --temacs=pbootstrap \
	--bin-dest /usr/local/bin/ --eln-dest /usr/local/lib/emacs/28.0.50/
Loading loadup.el (source)...
Dump mode: pbootstrap
Using load-path (/root/emacs/lisp /root/emacs/lisp/emacs-lisp /root/emacs/lisp/progmodes /root/emacs/lisp/language /root/emacs/lisp/international /root/emacs/lisp/textmodes /root/emacs/lisp/vc)
Loading emacs-lisp/byte-run...
Loading emacs-lisp/backquote...
Loading subr...
Loading version...
Loading emacs-lisp/emacs-ng...
Loading widget...
Loading custom...
Loading emacs-lisp/map-ynp...
Loading international/mule...
Loading international/mule-conf...
Loading env...
Loading format...
Loading bindings...
Loading window...
Loading files...
Loading emacs-lisp/macroexp...
Loading cus-face...
Loading faces...
Loading /root/emacs/lisp/loaddefs.el (source)...
Loading button...
Loading emacs-lisp/nadvice...
Loading emacs-lisp/cl-preloaded...
Loading obarray...
Loading abbrev...
Loading simple...
Loading help...
Loading jka-cmpr-hook...
Loading epa-hook...
Loading international/mule-cmds...
Loading case-table...
Loading /root/emacs/lisp/international/charprop.el (source)...
Loading international/characters...
Loading international/charscript...
Loading composite...
Loading language/chinese...
Loading language/cyrillic...
Loading language/indian...
Loading language/sinhala...
Loading language/english...
Loading language/ethiopic...
Loading language/european...
Loading language/czech...
Loading language/slovak...
Loading language/romanian...
Loading language/greek...
Loading language/hebrew...
Loading international/cp51932...
Loading international/eucjp-ms...
Loading language/japanese...
Loading language/korean...
Loading language/lao...
Loading language/tai-viet...
Loading language/thai...
Loading language/tibetan...
Loading language/vietnamese...
Loading language/misc-lang...
Loading language/utf-8-lang...
Loading language/georgian...
Loading language/khmer...
Loading language/burmese...
Loading language/cham...
Loading indent...
Loading emacs-lisp/cl-generic...
Loading minibuffer...
Loading frame...
Loading startup...
Loading term/tty-colors...
Loading font-core...
Loading emacs-lisp/syntax...
Loading font-lock...
Loading jit-lock...
Loading mouse...
Loading scroll-bar...
Loading select...
Loading emacs-lisp/timer...
Loading emacs-lisp/easymenu...
Loading isearch...
Loading rfn-eshadow...
Loading menu-bar...
Loading tab-bar...
Loading emacs-lisp/lisp...
Loading textmodes/page...
Loading register...
Loading textmodes/paragraphs...
Loading progmodes/prog-mode...
Loading emacs-lisp/lisp-mode...
Loading progmodes/elisp-mode...
Loading textmodes/text-mode...
Loading textmodes/fill...
Loading newcomment...
Loading replace...
Loading emacs-lisp/tabulated-list...
Loading buff-menu...
Loading fringe...
Loading emacs-lisp/regexp-opt...
Loading image...
Loading international/fontset...
Loading dnd...
Loading tool-bar...
Loading dynamic-setting...
Loading x-dnd...
Loading term/common-win...
Loading term/x-win...
Loading mwheel...
Loading emacs-lisp/float-sup...
Loading vc/vc-hooks...
Loading vc/ediff-hook...
Loading uniquify...
Loading electric...
Loading emacs-lisp/eldoc...
Loading cus-start...
Loading tooltip...
Loading international/iso-transl...
Loading /root/emacs/lisp/leim/leim-list.el (source)...
Finding pointers to doc strings...
Finding pointers to doc strings...done
Dumping under the name bootstrap-emacs.pdmp
Dumping fingerprint: ad5a6fac7645232849d26046436e231fee2f5b876a162b722105a8673cac407b

pdumper.c:1361: Emacs fatal error: assertion failed: already_dumped_object
Fatal error 6: Aborted
Backtrace:
./temacs[0x607b02]
./temacs[0x5d8a04]
./temacs[0x66d37b]
./temacs[0x66f998]
./temacs[0x670840]
./temacs[0x6708d9]
./temacs[0x671fcd]
./temacs[0x673d2c]
./temacs[0x674730]
./temacs[0x676d6e]
./temacs[0x69c0d7]
./temacs[0x696150]
./temacs[0x69bce6]
./temacs[0x6963bf]
./temacs[0x69843f]
./temacs[0x69bce6]
./temacs[0x698c9d]
./temacs[0x69bce6]
./temacs[0x6963bf]
./temacs[0x69843f]
./temacs[0x69bce6]
./temacs[0x6963bf]
./temacs[0x69843f]
./temacs[0x69bce6]
./temacs[0x696150]
./temacs[0x69bce6]
./temacs[0x6d4801]
./temacs[0x6d2661]
./temacs[0x69c18e]
./temacs[0x69b675]
./temacs[0x5ddc0d]
./temacs[0x69945f]
./temacs[0x5ddc51]
./temacs[0x6988c6]
./temacs[0x5ddb42]
./temacs[0x5dd133]
./temacs[0x5dd314]
./temacs[0x5db070]
/lib64/libc.so.6(__libc_start_main+0xd5)[0x7ffff6233b75]
./temacs[0x45986e]
make[1]: *** [Makefile:910: bootstrap-emacs.pdmp] Aborted (core dumped)
make[1]: *** Deleting file 'bootstrap-emacs.pdmp'
make[1]: Leaving directory '/root/emacs/src'
make: *** [Makefile:432: src] Error 2

</details>

cc: @DavidDeSimone @brotzeit (or maybe someone else?) have some suggestions to help me get the picture?

thanks!

### CODE REVIEW

**Comment by harryfei** [comment]
Rebase master please. There's some commit fixed this CI error.

**Review by brotzeit — APPROVED** [review]
LGTM

File: docker/ubuntu/config.yaml [hunk 1]
```diff
@@ -17,4 +17,5 @@ base_config: |
         libxml2-dev \
         libxpm-dev \
         libxt-dev \
-        texinfo
+        texinfo \
+        libgccjit-5-dev gcc-5 g++-5
```
Comment (black7375) [inline docker/ubuntu/config.yaml:21]: We are using `libgccjit-9-dev`,  `gcc-9`,  `g++-9` at build.

https://github.com/emacs-ng/emacs-ng/blob/bc889e9eebcc01937e05d559594cde1027cef479/build_emacs_ng.sh#L19

Is there any reason to use the lower version?

Comment (jman-schief) [reply docker/ubuntu/config.yaml:21]: @black7375 this is the GCC version availabe in Ubuntu Xenial (which this image is based on).

To be honest I don't have clear why emacs-ng needs these Docker images at all (based on ancient Ubuntu or 32bit Ubuntu versions)

Comment (black7375) [reply docker/ubuntu/config.yaml:21]: It could be because it was made a long time ago

**Pull request merged**